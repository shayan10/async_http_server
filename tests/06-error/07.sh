#! /bin/bash
# if post request is malformed -> 404

PORT=$@

EOL=$'\r\n'
DATA=$'This post is malformed!'
REQUEST1=$'POST /write HTTP/1.1\r
Content-Length: '${#DATA}${EOL}${DATA}
REQUEST2=$'GET /read HTTP/1.1'${EOL}${EOL}

for i in $(seq 1 3); do
    printf "$REQUEST1" | nc 127.0.0.1 $PORT
    printf "\n"
    printf "$REQUEST2" | nc 127.0.0.1 $PORT
done
