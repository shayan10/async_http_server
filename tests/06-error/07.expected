HTTP/1.1 400 Bad Request


HTTP/1.1 200 OK
Content-Length: 7

<empty>HTTP/1.1 400 Bad Request


HTTP/1.1 200 OK
Content-Length: 7

<empty>HTTP/1.1 400 Bad Request


HTTP/1.1 200 OK
Content-Length: 7

<empty>